<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>662</width>
    <height>568</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>压缩文件密码破解器</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>../res/icon.ico</normaloff>../res/icon.ico</iconset>
  </property>
  <property name="layoutDirection">
   <enum>Qt::LeftToRight</enum>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="1" column="0">
     <widget class="QTabWidget" name="dict_source">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="toolTip">
       <string>切换字典源</string>
      </property>
      <property name="statusTip">
       <string/>
      </property>
      <property name="whatsThis">
       <string>切换字典源</string>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="internal_dict">
       <attribute name="title">
        <string>使用内置字典</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="check_boxes">
          <item>
           <widget class="QCheckBox" name="checkBox_num">
            <property name="toolTip">
             <string>0123456789</string>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string>密码是否包含数字</string>
            </property>
            <property name="text">
             <string>数字
(0～9)</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_lower_letter">
            <property name="toolTip">
             <string>abcdefghijklmnopqrstuvwxyz</string>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string>密码是否包含小写字母</string>
            </property>
            <property name="text">
             <string>小写字母
(a～z)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_upper_letter">
            <property name="toolTip">
             <string>ABCDEFGHIJKLMNOPQRSTUVWXYZ</string>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string>密码是否包含大写字母</string>
            </property>
            <property name="text">
             <string>大写字母
(A～Z)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_symbols">
            <property name="toolTip">
             <string>!&quot;#$%&amp;'()*+,-./:;&lt;=&gt;?@[\]^_`{|}~</string>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string>密码是否包含特殊符号</string>
            </property>
            <property name="text">
             <string>特殊符号
(!&quot;#$%&amp;...)</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="digits_boxes">
          <item>
           <widget class="QSpinBox" name="digit_min">
            <property name="toolTip">
             <string>设置最低位数</string>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string>设置最低位数</string>
            </property>
            <property name="suffix">
             <string>位</string>
            </property>
            <property name="prefix">
             <string/>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>8</number>
            </property>
           </widget>
          </item>
          <item alignment="Qt::AlignHCenter">
           <widget class="QLabel" name="label_8">
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>40</height>
             </size>
            </property>
            <property name="text">
             <string>至</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="digit_max">
            <property name="toolTip">
             <string>设置最高位数</string>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string>设置最高位数</string>
            </property>
            <property name="suffix">
             <string>位</string>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>8</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="export_boxes">
          <item>
           <widget class="QPushButton" name="button_export">
            <property name="maximumSize">
             <size>
              <width>167</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>点击开始导出字典</string>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string>点击开始导出字典</string>
            </property>
            <property name="text">
             <string>导出字典</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QProgressBar" name="progress_export">
            <property name="toolTip">
             <string>导出字典进度</string>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string>导出字典进度</string>
            </property>
            <property name="value">
             <number>0</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="export_path">
            <property name="toolTip">
             <string>导出字典的保存路径</string>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string>导出字典的保存路径</string>
            </property>
            <property name="placeholderText">
             <string>在此键入导出字典的保存路径</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="button_export_path">
            <property name="toolTip">
             <string>点击选择导出字典的保存路径</string>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string>点击选择导出字典的保存路径</string>
            </property>
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="external_dict">
       <attribute name="title">
        <string>使用自定义字典</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="1">
         <widget class="QLineEdit" name="dict_path">
          <property name="toolTip">
           <string>自定义字典位置</string>
          </property>
          <property name="statusTip">
           <string/>
          </property>
          <property name="whatsThis">
           <string>自定义字典位置</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>请选择自定义字典</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QToolButton" name="button_dict_path">
          <property name="toolTip">
           <string>点击选择自定义字典</string>
          </property>
          <property name="statusTip">
           <string/>
          </property>
          <property name="whatsThis">
           <string>点击选择自定义字典</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="toolTip">
       <string>配置后台</string>
      </property>
      <property name="statusTip">
       <string/>
      </property>
      <property name="whatsThis">
       <string>配置后台</string>
      </property>
      <property name="title">
       <string>配置后台</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>使用核心数量</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLCDNumber" name="core_num">
           <property name="toolTip">
            <string>后台消费者的线程数或进程数</string>
           </property>
           <property name="statusTip">
            <string/>
           </property>
           <property name="whatsThis">
            <string>后台消费者的线程数或进程数</string>
           </property>
           <property name="autoFillBackground">
            <bool>true</bool>
           </property>
           <property name="frameShape">
            <enum>QFrame::Box</enum>
           </property>
           <property name="frameShadow">
            <enum>QFrame::Raised</enum>
           </property>
           <property name="segmentStyle">
            <enum>QLCDNumber::Filled</enum>
           </property>
           <property name="value" stdset="0">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="cpu_slider">
           <property name="toolTip">
            <string>滑动调节后台消费者的线程数或进程数</string>
           </property>
           <property name="statusTip">
            <string/>
           </property>
           <property name="whatsThis">
            <string>滑动调节后台消费者的线程数或进程数</string>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>密码批量处理数</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLCDNumber" name="batch_size">
           <property name="toolTip">
            <string>导出或导入密码时批量处理密码数量</string>
           </property>
           <property name="statusTip">
            <string/>
           </property>
           <property name="whatsThis">
            <string>导出或导入密码时批量处理密码数量</string>
           </property>
           <property name="intValue" stdset="0">
            <number>20000</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDial" name="dial">
           <property name="toolTip">
            <string>转动调节批量处理密码的数量</string>
           </property>
           <property name="statusTip">
            <string/>
           </property>
           <property name="whatsThis">
            <string>转动调节批量处理密码的数量</string>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>20000</number>
           </property>
           <property name="value">
            <number>20000</number>
           </property>
           <property name="wrapping">
            <bool>false</bool>
           </property>
           <property name="notchTarget">
            <double>3.700000000000000</double>
           </property>
           <property name="notchesVisible">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QGroupBox" name="groupBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="toolTip">
       <string>开始破解</string>
      </property>
      <property name="statusTip">
       <string/>
      </property>
      <property name="whatsThis">
       <string>开始破解</string>
      </property>
      <property name="title">
       <string>开始破解</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>请选择压缩文件</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="zipfile_path">
           <property name="toolTip">
            <string>压缩文件位置</string>
           </property>
           <property name="statusTip">
            <string/>
           </property>
           <property name="whatsThis">
            <string>压缩文件位置</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="button_zipfile_path">
           <property name="toolTip">
            <string>点击选择压缩文件</string>
           </property>
           <property name="statusTip">
            <string/>
           </property>
           <property name="whatsThis">
            <string>点击选择压缩文件</string>
           </property>
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>请选择解压位置</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="extract_path">
           <property name="toolTip">
            <string>解压路径</string>
           </property>
           <property name="statusTip">
            <string/>
           </property>
           <property name="whatsThis">
            <string>解压路径</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="button_extract_path">
           <property name="toolTip">
            <string>点击选择解压位置</string>
           </property>
           <property name="statusTip">
            <string/>
           </property>
           <property name="whatsThis">
            <string>点击选择解压位置</string>
           </property>
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item alignment="Qt::AlignLeft">
          <widget class="QPushButton" name="pushButton_crack">
           <property name="toolTip">
            <string>点击开始破解</string>
           </property>
           <property name="statusTip">
            <string/>
           </property>
           <property name="whatsThis">
            <string>点击开始破解</string>
           </property>
           <property name="text">
            <string>开始破解</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QProgressBar" name="progress_crack">
           <property name="toolTip">
            <string>破解进度</string>
           </property>
           <property name="statusTip">
            <string/>
           </property>
           <property name="whatsThis">
            <string>破解进度</string>
           </property>
           <property name="value">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item alignment="Qt::AlignLeft">
          <widget class="QLabel" name="label_password_is">
           <property name="text">
            <string>密码是</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="password">
           <property name="toolTip">
            <string>破解出的压缩密码</string>
           </property>
           <property name="statusTip">
            <string/>
           </property>
           <property name="whatsThis">
            <string>破解出的压缩密码</string>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item row="4" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton">
        <property name="text">
         <string>关于软件</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_2">
        <property name="text">
         <string>退出</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>cpu_slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>core_num</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>347</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>169</x>
     <y>61</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_export_path</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>select_export_path()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>632</x>
     <y>295</y>
    </hint>
    <hint type="destinationlabel">
     <x>657</x>
     <y>220</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_export</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_export_dict()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>64</x>
     <y>305</y>
    </hint>
    <hint type="destinationlabel">
     <x>7</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dial</sender>
   <signal>valueChanged(int)</signal>
   <receiver>batch_size</receiver>
   <slot>display(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>584</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>70</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>progress_export</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_export_progress_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>181</x>
     <y>302</y>
    </hint>
    <hint type="destinationlabel">
     <x>9</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_zipfile_path</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>select_zipfile_path()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>632</x>
     <y>379</y>
    </hint>
    <hint type="destinationlabel">
     <x>656</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_extract_path</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>select_extract_path()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>616</x>
     <y>415</y>
    </hint>
    <hint type="destinationlabel">
     <x>658</x>
     <y>442</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>623</x>
     <y>508</y>
    </hint>
    <hint type="destinationlabel">
     <x>662</x>
     <y>514</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_dict_path</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>select_dict_path()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>609</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>655</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_crack</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_crack_password()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>82</x>
     <y>461</y>
    </hint>
    <hint type="destinationlabel">
     <x>82</x>
     <y>503</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>progress_crack</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_crack_progress_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>461</y>
    </hint>
    <hint type="destinationlabel">
     <x>153</x>
     <y>502</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>489</x>
     <y>523</y>
    </hint>
    <hint type="destinationlabel">
     <x>451</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>validate_bool()</slot>
  <slot>select_export_path()</slot>
  <slot>on_export_dict()</slot>
  <slot>on_export_progress_changed(int)</slot>
  <slot>select_zipfile_path()</slot>
  <slot>select_extract_path()</slot>
  <slot>select_dict_path()</slot>
  <slot>on_crack_password()</slot>
  <slot>on_crack_progress_changed(int)</slot>
  <slot>on_about()</slot>
 </slots>
</ui>
